nexus/sdk-go.md

```diff
@@ -0,0 +1,352 @@
+) nexus.Operation[I, O]
+```
+
+> NOTE: this is a simple wrapper around the Nexus Go SDK's
+> [NewSyncOperation](https://pkg.go.dev/github.com/nexus-rpc/sdk-go@v0.0.7/nexus#NewSyncOperation), which can be used in
+> sync operations that don't require a Temporal client.
+
+**Usage**:
+
+```go
+import (
+	"github.com/nexus-rpc/sdk-go/nexus"
+	"go.temporal.io/sdk/client"
+	"go.temporal.io/sdk/temporalnexus"
+)
+
+opGetStatus := temporalnexus.NewSyncOperation("get-status", func(ctx context.Context, c client.Client, id string, opts nexus.StartOperationOptions) (int, error) {
+	res, err := c.QueryWorkflow(ctx, id, "", "some-query", nil)
+	if err != nil {
+		return 0, err
+	}
+	var ret int
+	return ret, res.Get(&ret)
+})
+
+// Operations don't have to return values.
+opContinue := temporalnexus.NewSyncOperation("continue-processing", func(ctx context.Context, c client.Client, id string, opts nexus.StartOperationOptions) (nexus.NoValue, error) {
+	return nil, c.SignalWorkflow(ctx, id, "", "continue-processing", nil)
+})
+```
+
+### NewWorkflowRunOperation
+
+```go
+package temporalnexus
+
+import (
+	"github.com/nexus-rpc/sdk-go/nexus"
+	"go.temporal.io/sdk/client"
+)
+
+// NewWorkflowRunOperation map an operation to a workflow run.
+func NewWorkflowRunOperation[I, O any](
+	name string,
+	workflow func(internal.Context, I) (O, error),
+	getOptions func(context.Context, I, nexus.StartOperationOptions) (client.StartWorkflowOptions, error),
+) nexus.Operation[I, O]
+
+type WorkflowRunOperationOptions[I, O any] struct {
+	// Operation name.
+	Name string
+	// Workflow function to map this operation to. The operation input maps directly to workflow input.
+	// The workflow name is resolved as it would when using this function in client.ExecuteOperation.
+	// OptionsGetter must be provided when setting this option. Mutually exclusive with Handler.
+	Workflow func(internal.Context, I) (O, error)
+	// Options for starting the workflow. Must be set if Workflow is set. Mutually exclusive with Handler.
+	OptionsGetter func(context.Context, I, nexus.StartOperationOptions) (client.StartWorkflowOptions, error)
+	// Handler for starting a workflow with a different input than the operation. Mutually exclusive with Workflow.
+	Handler func(context.Context, I, nexus.StartOperationOptions) (WorkflowHandle[O], error)
+}
+
+// NewWorkflowRunOperation map an operation to a workflow run with the given options.
+func NewWorkflowRunOperationWithOptions[I, O any](opts WorkflowRunOperationOptions[I, O]) nexus.Operation[I, O]
+
+type WorkflowHandle[T any] interface {
+	ID() string
+	RunID() string
+}
+
+// ExecuteWorkflow starts a workflow run, linking the execution chain to a Nexus operation (subsequent runs started from
+// continue-as-new and retries).
+// Automatically propagates the callback and request ID from the nexus options to the workflow.
+func ExecuteWorkflow[I, O any, WF func(internal.Context, I) (O, error)](
```

Comment:
The purpose of this is a bit confusing, can you clarify this is meant to be used in a handler?

Reply:
Yes.

Reply:
Done.